source/common/protobuf/utility.cc

```diff
@@ -171,7 +173,14 @@ void MessageUtil::checkForDeprecation(const Protobuf::Message& message, Runtime:
           "for details.",
           field->full_name(), filename);
       if (warn_only) {
-        ENVOY_LOG_MISC(warn, "{}", err);
+        double log_delay;
+        time_t current_time = time(nullptr);
+        log_delay = difftime(current_time, last_log_time);
+        //It ensures that at least 30s is elapsed between two logs.
```

Comment:
Thanks for tackling this!  Unfortunately this has the problem that if you are using multiple deprecated fields, you will not get warnings about them all.  I think to actually get folks to migrate off of all their deprecated fields there will need to be some state tracking of which field is being checked.